kinesis/kinesis.go

```diff
@@ -43,7 +43,7 @@ const (
 const (
 	// Kinesis API Limit https://docs.aws.amazon.com/sdk-for-go/api/service/kinesis/#Kinesis.PutRecords
 	maximumRecordsPerPut      = 500
-	maximumPutRecordBatchSize = 1024 * 1024 * 5 // 5 MB
+	maximumPutRecordBatchSize = 1024 * 1024 * 5 - partitionKeyMaxLength // 5 MB
```

Comment:
I'm not sure this is the right place to fix this. I think that the `maximumPutRecordBatchSize` should remain the same and that the partition keys need to be properly involved in the payload size calculated in AddRecord.

Am I misunderstanding this?

@hossain-rayhan please take a look.

Reply:
I agree. This is the simplest workaround which additionally wastes 194 characters, in a case when partitionKey is generated by randomString().

Reply:
Yes. If user doesn't mention a partition key we generate a random one of length 8 (chars). We can keep this default size unchanged, and adjust the `maximumPutRecordBatchSize` where checking for the `partition_key` value. It will save some spaces.

Also, thanks @falaa for the great finding.

Reply:
But the partition key is part of every record in the API request: https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html

And this only subtracts it once... so I think the math doesn't work..

Reply:
@falaa this is bug though- can you open an issue for it as well?

Reply:
@PettitWesley you are right. We should update the calculation inside the AddRecord() function.